### PR TITLE
Add OAuth 2.0 client for NAVER

### PR DESCRIPTION
**Pull request by joonas-yoon**
created new client for NAVER.

- [x] Add Client
- [x] Update documentation
- [x] Add PyTest

for information, [NAVER](https://www.navercorp.com/en/service/featured) is a one of famous web service for search in South Korea.

### CODE REVIEW

**Comment by frankie567** [comment]
Looks good! I see that `BASE_SCOPE` is empty, is it normal? No scope are required to access the API?

File: httpx_oauth/clients/naver.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            ACCESS_TOKEN_ENDPOINT,
+            refresh_token_endpoint=REFRESH_TOKEN_ENDPOINT,
+            revoke_token_endpoint=REVOKE_TOKEN_ENDPOINT,
+            name=name,
+            base_scopes=scopes,
+        )
+
+    async def revoke_token(self, token: str, token_type_hint: str = None) -> None:
+        async with self.get_httpx_client() as client:
+            data = {
```
Comment (frankie567) [inline httpx_oauth/clients/naver.py:50]: `revoke_token` should be covered by a test. You can draw inspiration from the base one: https://github.com/frankie567/httpx-oauth/blob/28d090606d2b9f56123e8bd7bad624dbd0fe7120/tests/test_oauth2.py#L186-L218

**Comment by joonas-yoon** [comment]
> Looks good! I see that `BASE_SCOPE` is empty, is it normal? No scope are required to access the API?

Yes it's right what you see. According to official API documentation, there is no scope to get an access token.

**Comment by joonas-yoon** [comment]
got 100% coverage! and fix typo on test class name

**Comment by frankie567** [comment]
Looks good, thanks! Happy to have Korean services available 😄🇰🇷

**Pull request merged**